### ISSUE TITLE
Implement 'P0067R5 Elementary string conversions'

### ISSUE THREAD

**Issue by cjdb** [llvm/llvm-project#99940]

**Comment by h-vetinari** [comment]
I'm glad there are finally [issues](https://github.com/llvm/llvm-project/issues/79219) for these topics; that makes it much easier to collect publicly available information about them, rather have them scattered in the heads of core libcxx contributors.

* #56261
* #59374
* #64180
* #92711

Also some context I had kept from long [ago](https://github.com/llvm/llvm-project/issues/59374#issuecomment-1469496870)

> @h-vetinari: Stumbled over this while looking for the status of P0067R5; took me a while to unearth some old comments on this topic in https://reviews.llvm.org/D70631:
> 
> > @h-vetinari: I was under the [impression](https://twitter.com/lemire/status/1417215574657179657) that https://github.com/llvm/llvm-project/commit/87c016078ad72c46505461e4ff8bfa04819fe7ba did most of the work for from_chars?
> 
> > @mordante: Thanks for both links I wasn't aware of this effort!
> > At the moment I'm too busy with `<format>` to look at `from_chars`, but this is useful when somebody starts working on `from_chars`. (I might look at it when I finished the larger projects I'm working on.)

**Comment by lemire** [comment]
The `from_chars` issue could be largely solved with https://github.com/fastfloat/fast_float 

It is what GCC/glibc++ does.

I also left a comment in https://github.com/llvm/llvm-project/issues/92711

I'd be happy to help.

**Comment by h-vetinari** [comment]
The implementation of the remaining bits landed in #91651, which completes this paper except for long-double support (needs #113237). Feature test macro wasn't set yet.

**Comment by frederick-vs-ja** [comment]
Looks like that we're now very close to setting FTM on `long double`-eq-`double` platforms.

**Comment by cor3ntin** [comment]
@frederick-vs-ja I'm trying to figure out what the status of that is since you closed your pr. Is it actually complete  ? Thanks!

**Comment by philnik777** [comment]
@cor3ntin No, it's not complete. There is no fundamental hurdle to adding the `long double` versions (except on some funky IBM system), but someone has to write the tests for it.

**Comment by mordante** [comment]
> but someone has to write the tests for it.

There is some implementation work to do too, especially for `to_chars`. Just to add a bit more information, adding the tests is no small effort. (I expect it to take over a week of work.)
